src/core/src/CoreMain.py

```diff
@@ -101,6 +101,7 @@ def __init__(self, argv):
                     patch_installer.mark_installation_completed()
                     overall_patch_installation_operation_successful = True
                 self.update_patch_substatus_if_pending(patch_operation_requested, overall_patch_installation_operation_successful, patch_assessment_successful, configure_patching_successful, status_handler, composite_logger)
+            status_handler.log_truncated_removed_packages_if_any()
 
```

Comment:
Status truncation should happen at every status file write and not just at the end of a patch operation. Reason for this as we have discussed before is, the agent reads status file at regular intervals and reports it back to CRP. If the status file exceeds limit during any of these times, agent will truncate it and report the broken json